src/crud/document.py

```
@@ -25,14 +25,34 @@ async def query_documents(
+    Returns:
+        Sequence of matching documents
+    """
+    # Use provided embedding or generate one
+    if embedding is None:
+        # Using ModelClient for embeddings
+        try:
+            embedding = await embedding_client.embed(query)
+        except ValueError as e:
+            raise ValidationException(
+                f"Query exceeds maximum token limit of {settings.MAX_EMBEDDING_TOKENS}."
+            ) from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)